api/views/share.py

```diff
@@ -107,6 +108,15 @@ def get(self, request, *args, **kwargs):
         return Response(ser.data)
 
 
+def user_favicon_view(request, username):
+    user = ShareUser.objects.get(username=username)
```

Comment:
Looks like this guy 500's rather than 404's if it fails

This should also only respond to GETs, just for semantic correctness.